etna/ciim/models.py

```diff
@@ -181,9 +163,6 @@ def count(self):
         self._fetch(start=0, size=0)
         return self._response["hits"]["total"]["value"]
 
-    def __iter__(self):
-        return SearchQueryIterator(self)
-
     def first(self):
         """Return the first item in resultset without or None if nothing is found."""
         try:
```

Comment:
Not sure why do we rely on exceptions to return, can we do something like this if self[0] is None: return None